src/react/components/APILink.js

```diff
@@ -1,45 +1,93 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import qs from 'querystringify';
+import { connect } from 'react-redux';
 import './APILink.scss';
 
 const propTypes = {
   disabled: PropTypes.bool,
   isCsLink: PropTypes.bool,
   name: PropTypes.string,
+  showNewWindow: PropTypes.bool,
   title: PropTypes.string,
   ucFrom: PropTypes.string,
   ucFromLink: PropTypes.string,
   ucFromText: PropTypes.string,
+  ucUpdateCache: PropTypes.string,
   url: PropTypes.string,
   urlId: PropTypes.string,
 };
 
-const APILink = ({ disabled, title, name, url, ucFrom, ucFromText }) => {
+const APILink = ({
+  disabled,
+  showNewWindow,
+  title,
+  name,
+  url,
+  ucFromLink,
+  ucFromText,
+  ucUpdateCache,
+}) => {
+  if (!url) {
+    // TODO - how to handle when Link API is not coming from CS?
+    // eslint-disable-next-line no-console
+    console.log(
+      '**URL not found** Component could not be rendered.  Check Link API data.'
+    );
+    return null;
+  }
   if (disabled) {
     return <span className="APILink APILink--disabled">{name}</span>;
   }
 
-  const currentUrl = window.location.href;
   const queryStringPrefix = url.includes('?') ? '&' : true;
-  const href =
-    url +
-    qs.stringify(
-      {
-        ucFrom,
-        ucFromText,
-        ucFromLink: currentUrl,
-      },
-      queryStringPrefix
-    );
+  const queryStringPrefixFromLink = ucFromLink.includes('?') ? '&' : true;
+  // the ucFromLink value may need to have a 'ucUpdateCache' parameter appended to it,
+  // if we get the value from the Link API the following will append it to the ucFromLink
+  const ucFromLinkWithParameter = ucUpdateCache
+    ? ucFromLink +
+      qs.stringify({ ucUpdateCache: ucUpdateCache }, queryStringPrefixFromLink)
+    : ucFromLink;
```

Comment:
This is complicated with all the long names, why don't we extract a new function and call it something like `returnUrl`?